python/cudf/cudf/core/dataframe.py

```diff
@@ -3609,11 +3610,13 @@ def agg(self, aggs, axis=None):
           * Not supporting: ``axis``, ``*args``, ``**kwargs``
 
         """
-        # TODO: Remove the typecasting below once issue #6846 is fixed
-        # link <https://github.com/rapidsai/cudf/issues/6846>
         dtypes = [self[col].dtype for col in self._column_names]
         common_dtype = find_common_type(dtypes)
-        df_normalized = self.astype(common_dtype)
+        if not is_bool_dtype(common_dtype) and any(
+            is_bool_dtype(dtype) for dtype in dtypes
+        ):
+            raise MixedTypeError("Cannot create a column with mixed types")
+        df_normalized = self.copy()
```

Comment:
Curious why we need the copy here? Is it because a UDF might modify self?

Reply:
Nope. It's a stale change, removed it.